### ISSUE TITLE
Add tls.client_metadata_key support to vector source (and maybe http_server as well)

### ISSUE THREAD

**Issue by vbr-cts** [vectordotdev/vector#20477]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

When sending logs between vector instances, it is currently possible to use TLS mutual authentication. The client and server will verify that they each have valid certificates. However, no information about this authentication is available for routing, remaping, or lua processing.

Having that information would mean we could then do authorization, or authenticate and track message origin.

### Attempted Solutions

The `socket` source supports a `tls.client_metadata_key` option, that lets me do exactly what I need. However it is

### Proposal

Add support for the `tls.client_metadata_key` option to the vector source. Ideally, it would also be added to the `http_server` source as well. 

### Version

vector 0.38.0 (x86_64-unknown-linux-gnu ea0ec6f 2024-05-07 14:34:39.794027186)

**Comment by ypid-work** [comment]
> Ideally, it would also be added to the http_server source as well.

#13596 already covers this.

**Comment by jszwedko** [comment]
> > Ideally, it would also be added to the http_server source as well.
> 
> #13596 already covers this.

Good call, I think https://github.com/vectordotdev/vector/issues/13596 covers this issue so I'll closet his one as a duplicate.

**Comment by ypid-work** [comment]
@jszwedko This issue does not cover the vector source in which I am very interested in. This issue should be reopened and limited to the Vector source or this issue should fully include HTTP source as requirement and then #13596 can be closed.
